examples/medplum-chart-demo/src/bots/core/bot-utils.ts

```diff
@@ -0,0 +1,271 @@
+
+export interface ObstetricObservationData extends ObservationData {
+  gravida?: number;
+  para?: number;
+  gestationalDays?: number;
+  gestationalWeeks?: number;
+  totalWeightGain?: Quantity;
+}
+
+export interface ConditionData {
+  reasonForVisit: Coding;
+  problemList: boolean;
+}
+
+export interface ClinicalImpressionData {
+  visitLength?: number;
+  assessment?: string;
+}
+
+export function createObservationEntries(
+  observationData: GeneralObservationData | GynecologyObservationData | ObstetricObservationData,
+  encounter: Encounter,
+  user: Practitioner,
+  createEntryFunction: (
+    key: string,
+    request: BundleEntryRequest,
+    genericObservation: Observation,
+    observationData: GeneralObservationData | GynecologyObservationData | ObstetricObservationData
+  ) => BundleEntry
+): BundleEntry[] {
+  const entries: BundleEntry[] = [];
+  const request: BundleEntryRequest = { method: 'POST', url: 'Observation' };
+  // Create a generic observation that just needs to have a code and value added to it
+  const genericObservation: Observation = {
+    resourceType: 'Observation',
+    status: 'preliminary',
+    subject: encounter.subject,
+    encounter: { reference: getReferenceString(encounter) },
+    performer: [{ reference: getReferenceString(user) }],
+    effectiveDateTime: observationData.date,
+    code: {},
+  };
+
+  // Loop over each entry of the observation data
+  for (const [key, value] of Object.entries(observationData)) {
+    // If there is no value for a key, skip
+    // Check that blood pressure has a measurement for at least one of systolic and diastolic, otherwise skip
+    // Date is used to get the time for each observation, but does not create its own observation so skip it
+    if (!value || (key === 'bloodPressure' && !value.systolic && !value.diastolic) || key === 'date') {
+      continue;
+    }
+
+    // Create an entry and add it to the entries array
+    entries.push(createEntryFunction(key, request, genericObservation, observationData));
+  }
+
+  return entries;
+}
+
+export function handleBloodPressure(observationData: ObservationData): ObservationComponent[] {
+  const components: ObservationComponent[] = [];
+  const bloodPressure = observationData.bloodPressure;
+
+  // If a diastolic measurement exists, add it
+  if (bloodPressure.diastolic) {
+    components.push({
+      code: { coding: [{ code: '8462-4', system: 'http://loinc.org', display: 'Diastolic blood pressure' }] },
+      valueQuantity: {
+        value: bloodPressure.diastolic,
+        unit: 'mm[Hg]',
+      },
+    });
+  }
+
+  // If a systolic measurement exists, add it
+  if (bloodPressure.systolic) {
+    components.push({
+      code: { coding: [{ code: '8480-6', system: 'http://loinc.org', display: 'Systolic blood pressure' }] },
+      valueQuantity: {
+        value: bloodPressure.systolic,
+        unit: 'mm[Hg]',
+      },
+    });
+  }
+
+  return components;
+}
+
+export function createConditionEntries(
```

Comment:
Add a JSDoc comment for this, and link to your document called "Representing Diagnoses"